addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/AWSHelper.java

```diff
@@ -92,6 +100,21 @@ public static AWSHelper getInstance() {
         return INSTANCE;
     }
 
+    /**
+     * A wrapper for enrichment that handles errors and reduces boilerplate code.
+     */
+    public static Collection<EnrichmentMetadata> enrich(Consumer<List<EnrichmentMetadata>> enricher) {
```

Comment:
I am open to suggestions about the design of this!  I basically wanted to wrap the call in a single exception handler and reduce the duplicate code.

Reply:
I think the duplication of the array allocation in each caller isn't important if it makes the code cleaner. So I'd do in the caller:

    return AWSHelper.enrich(() -> {
        List<EnrichmentMetadata> enriched = new ArrayList<>();
        // ... fill it ...
        return enriched;
    });

Which means this method would do:

    public static Collection<EnrichmentMetadata> enrich(Supplier<Collection<EnrichmentMetadata>> enricher) {
        try {
            return enricher.get();
        } catch (AmazonServiceException e) {
            // ... etc ...
        }	
    }

And actually rename this `enrich` method to something like `catchingExceptions` which is all it does. Which actually means it can be generic:

    public static <T> T catchingExceptions(Supplier<T> supplier) {
        try {
            return supplier.get();
        } catch (AmazonServiceException e) {
            // ... etc ...
        }	
    }

And you can also have a variant that returns `void` and takes just a `Runnable`, if you need to catch exceptions on runnables elsewhere in your code.

Reply:
:+1: